tests/doghouse/test_watch_persistence.py

```
@@ -0,0 +1,138 @@
+Verifies that repeated identical polls do not create duplicate snapshots,
+and that meaningful transitions do get persisted.
+"""
+import datetime
+from unittest.mock import MagicMock
+from doghouse.core.domain.blocker import Blocker, BlockerType, BlockerSeverity
+from doghouse.core.domain.snapshot import Snapshot
+from doghouse.core.services.recorder_service import RecorderService
+from doghouse.core.services.delta_engine import DeltaEngine
+
+
+def _make_service(
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `_make_service`

(ANN202)